### ISSUE TITLE
Serialization error on protobuf schema stream when schema full name contains references that are not nested

### ISSUE THREAD

**Issue by yalfah** [confluentinc/ksql#9598]
**Describe the bug**
KSQL fails to serialize data from topic onto stream that is created with Schema id and Schema id 

**To Reproduce**
Steps to reproduce the behavior, include:
1. The version of KSQL. (cloud KSQL)
2. Create a stream using schema id and schema full name:
`CREATE STREAM "121693_test_schemas" 
WITH (
  KAFKA_TOPIC='multiple_schemas_proto_test',
  VALUE_SCHEMA_ID=100273,PARTITIONS=1,
  VALUE_SCHEMA_FULL_NAME='LocationDeleteRequest',
  VALUE_FORMAT='PROTOBUF'); `

3. attempt to produce data with expected format such as through a basic insert into command fails:
`INSERT INTO `multiple_schemas_proto_test` (`request_id`, `location`) VALUES (struct(`id` := 'test1'), struct(`id` := 'test11'));`

**Expected behavior**
Stream defined with schema id and name should be able to deserialize using non nested referenced schemas on the same schema id

**Actual behaviour**
no data makes it onto the stream with serialization error:
`Could not serialize valueError serializing message to topic:<topicName> Failed to serialize Protobuf data from topic`

**Additional context**

example non-working schema:
```
syntax = "proto3";
package test.schemas.v1;

message LocationIdentifier {
  string id = 1;
}
message DeleteRequestIdentifier {
  string id = 1;
}
message LocationDeleteRequest {
  DeleteRequestIdentifier request_id = 1;
  LocationIdentifier location = 2;
}
```

nesting the schemas allows the data to be inserted in the same method described above:
```
syntax = "proto3";
package test.schemas.v1;

message LocationDeleteRequest {
  DeleteRequestIdentifier request_id = 1;
  LocationIdentifier location = 2;

  message LocationIdentifier {
    string id = 1;
  }
  message DeleteRequestIdentifier {
    string id = 1;
  }
}
```

**Comment by jnh5y** [comment]
@yalfah noted that https://github.com/confluentinc/ksql/pull/9369 may be related.

**Comment by spena** [comment]
@yalfah The issue is fixed in the next 0.29 release in https://github.com/confluentinc/ksql/pull/9369. 
```
ksql> create stream test with (kafka_topic='test', value_format='protobuf', value_schema_full_name='LocationDeleteRequest');

 Message        
----------------
 Stream created 
----------------


ksql> insert into test(request_id, location) VALUES (struct(id := 'test1'), struct(id := 'test11'));

ksql> select * from test;
+-----------------------------------------------------------------+-----------------------------------------------------------------+
|REQUEST_ID                                                       |LOCATION                                                         |
+-----------------------------------------------------------------+-----------------------------------------------------------------+
|{ID=test1}                                                       |{ID=test11}                                                      |
Query Completed
```

One issue you'll find, though, is that the SR schema columns must be in uppercase. You'll get another type of serialization error otherwise.
```
ksql> insert into test(request_id, location) VALUES (struct(id := 'test1'), struct(id := 'test11'));
Failed to insert values into 'TEST'. Could not serialize valueError serializing message to topic: test. Schema from Schema Registry mis
ses field with name: REQUEST_ID
```

**Comment by spena** [comment]
This is the issue related to the INSERT when columns are not defined in uppercase in SR.
https://github.com/confluentinc/ksql/issues/8997

**Comment by spena** [comment]
I'll close this issue. 
1) The reported issue is fixed on 0.29 - https://github.com/confluentinc/ksql/pull/9369
2) The INSERT Issue is reported on https://github.com/confluentinc/ksql/issues/8997
